sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/TenantProviderIdentifier.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// The identifier of a Provider from Tenant.
+    /// </summary>
+    public class TenantProviderIdentifier : ResourceIdentifier
```

Comment:
I would argue that this should extend TenantResourceIdentifier, as all concrete resource ids currently do.